src/mon/MgrMonitor.cc

```diff
@@ -1319,6 +1319,34 @@ bool MgrMonitor::prepare_command(MonOpRequestRef op)
       ss << "module '" << module << "' is not enabled";
     }
     pending_map.modules.erase(module);
+  } else if (prefix == "mgr module set queue threshold") {
+    string module;
+    int64_t threshold;
+    cmd_getval(cmdmap, "module", module);
+    cmd_getval(cmdmap, "threshold", threshold);
+    if (module.empty()) {
+      r = -EINVAL;
+      goto out;
+    }
+    if (pending_map.get_module_threshold_value(module)) {
+      pending_map.modules_threshold_value[module] = threshold;
+    } else {
+      pending_map.modules_threshold_value.insert({module, threshold});
+    }
+    ss << "module '" << module << "' threshold value is set to " << pending_map.get_module_threshold_value(module);
+  } else if (prefix == "mgr module unset queue threshold") {
+    string module;
+    cmd_getval(cmdmap, "module", module);
+    if (module.empty()) {
+      r = -EINVAL;
+      goto out;
+    }
+    if (!pending_map.get_module_threshold_value(module)) {
+      ss << "module '" << module << "' threshold value is not set ";
+      goto out;
+    }   
+    pending_map.modules_threshold_value.erase(module);
+    ss << "module '" << module << "' threshold value has been unset ";
```

Comment:
There's no need to make this part of the mgrmap. I think you can use module configs like

`ceph config set mgr mgr/volumes/max_command_queue_length 128`

Reply:
IIUC, you mean to say, instead of having a new command to set the threshold value, we should use the module config options. But as said here: https://tracker.ceph.com/issues/62712?issue_count=18&issue_position=1&next_issue_id=62537#note-9, this needs to be generic..Is it possible to have a single config option for all modules?

Reply:
You can have the plugin set its limit in response to the config change or look it up dynamically on the C++ side.

Reply:
> There's no need to make this part of the mgrmap. I think you can use module configs like
> 
> `ceph config set mgr mgr/volumes/max_command_queue_length 128`

This involves adding the option in the module.py of all the modules and then fetch the value whenever the check is required. Does it make sense to have the same config in each and every modules like these: https://github.com/ceph/ceph/blob/main/src/pybind/mgr/volumes/module.py#L527 or Am I misunderstanding it?

Reply:
No,  you would put it here:

https://github.com/ceph/ceph/blob/398563d959f96017fda47be5fac9d0a84c84ec1e/src/pybind/mgr/mgr_module.py#L920-L921